src/shared.js

```diff
@@ -1583,6 +1585,38 @@ export const updateCollectionSettingData = async (biospecimenData, tubes, partic
             }
         });
     }
+    else if (settings[visit][conceptIds.baseline.bloodCollectedTime] !== '' ||  settings[visit][conceptIds.clinicalDashboard.bloodCollectedTime] !== ''){
+        const participantBloodCollected = participantData[conceptIds.baseline.bloodCollected] === conceptIds.yes;
+        const collectionSetting = biospecimenData[conceptIds.collection.collectionSetting];
+        const isResearch = collectionSetting === conceptIds.research;
+        const isClinical = collectionSetting === conceptIds.clinical;
+        let totalBloodTubesAvail = 6; // 6 blood tubes are available. Update this number if more blood tubes are added
+        bloodTubes.forEach(tube => {
+            const tubeIsNotCollected = biospecimenData[tube.concept][conceptIds.collection.tube.isCollected] === conceptIds.no;
+            if (tubeIsNotCollected && participantBloodCollected) {
+                totalBloodTubesAvail -= 1 // counter keeps track of unchecked blood tubes
```

Comment:
I feel like all this logic could be more easily implemented if you did a `filter` on bloodTubes where `biospecimen[tube.concept][conceptIds.collection.tube.isCollected] === conceptIds.no;`

If that filter doesn't have length, we know to do the additional logic. Keeping track of a counter seems like extra work.